pandas/tests/indexes/test_numpy_compat.py

```diff
@@ -49,14 +50,21 @@ def test_numpy_ufuncs_basic(index, func):
         with tm.external_error_raised((TypeError, AttributeError)):
             with np.errstate(all="ignore"):
                 func(index)
-    elif isinstance(index, NumericIndex):
+    elif isinstance(index, NumericIndex) or (
+        not isinstance(index.dtype, np.dtype) and index.dtype._is_numeric
```

Comment:
Could we use something like `is_numeric_dtype(index.dtype)` for checking this instead? (don't know if it is exactly equivalent)

Reply:
im wary of that approach for two reasons: 1) i also have a hard time remembering which is_foo_dtype functions do something complicated (looks like is_bool_dtype is the main one) and 2) this is more explicit about intentionally allowing numeric EA dtypes